internal/backend/remote-state/s3/backend.go

```diff
@@ -772,9 +772,9 @@ func (b *Backend) Configure(obj cty.Value) tfdiags.Diagnostics {
 	}
 
 	if value := obj.GetAttr("assume_role"); !value.IsNull() {
-		cfg.AssumeRole = configureNestedAssumeRole(obj)
+		cfg.AssumeRole = append(cfg.AssumeRole, configureNestedAssumeRole(obj))
```

Comment:
The append here is a bit strange, as we know the field is not set.

IMO it would be more clear to create a new list inline.

Reply:
Just to be clear, you mean something like
```
	var roles []awsbase.AssumeRole
	if value := obj.GetAttr("assume_role"); !value.IsNull() {
		roles = append(roles, configureNestedAssumeRole(obj))
	} else if value := obj.GetAttr("role_arn"); !value.IsNull() {
		roles = append(roles, configureAssumeRole(obj))
	}
	cfg.AssumeRole = roles
```
If yes, this was my first approach, but functionally, it's the same, but more verbose.

Or you mean something like this?
```
	if value := obj.GetAttr("assume_role"); !value.IsNull() {
		cfg.AssumeRole = []awsbase.AssumeRole{configureNestedAssumeRole(obj)}
	} else if value := obj.GetAttr("role_arn"); !value.IsNull() {
		cfg.AssumeRole = []awsbase.AssumeRole{configureAssumeRole(obj)}
	}
```

Reply:
Something like your second example.  The code as written will only honor the first one it sees.  If we want to change that, we should address that in it's own PR as it will also require some documentation changes.

Reply:
Done

Reply:
Isn't it done this way because `role_arn` is deprecated? Meaning that `assume_role` is having higher priority in processing here?

<img width="831" alt="Screenshot 2025-02-18 at 15 02 46" src="https://github.com/user-attachments/assets/cfe85309-b800-4fd2-82de-5a47db6fc436" />

Reply:
That's correct, I should have double checked the docs :)

Reply:
Glad that this is cleared out. Better to leave a clean trace behind :D